### ISSUE TITLE
Impossible complete copy of files inside a note

### ISSUE THREAD

**Issue by fanofjop** [laurent22/joplin#8129]
## Environment :
 Joplin version: all of them since one year (today same thing with the latest : 2.10.16)
OS : windows (7 and +), Linux-Kubuntu (previous and last one : 23.04)

## The thing to know :
When a file is inserted into a note there are 3 things : 1) the displayed text, on the note (the name of the file), 2) the text of the tooltip (when the mouse is let above the displayed text) and of course 3) the url of the file.

## Steps to reproduce

1.  I select with the mouse the entire displayed text of a file previously inserted into a note. The format of the file has no importance (jpg, pdf, mp4, mp3, odt, docx, etc.).
![image](https://user-images.githubusercontent.com/107914828/236201007-8ba7431a-6c39-4aec-a727-bbff047277b2.png)
3. I make a copy (right clic) 
![image](https://user-images.githubusercontent.com/107914828/236201139-74d3b985-d814-4dbd-931c-de6f7f5cc8b2.png)
4. I paste the file
![image](https://user-images.githubusercontent.com/107914828/236201307-6c64b8cc-976b-4b20-bc45-88e835aae814.png)
5. Only the displayed text is copied. Not the URL and not the tooltip.
 So, of course, you can't open (link broken) the file you've just copied.

BUT (tip to avoid the issue) .............

If I put a little piece of text BEFORE or AFTER (just a character for example) the inserted file and copy/paste the whole thing : it works !!!

I select and copy :
![image](https://user-images.githubusercontent.com/107914828/236202538-d83a3357-c651-4d28-87ba-1bba0d1cf2a4.png)

I paste :
![image](https://user-images.githubusercontent.com/107914828/236202762-7a99d066-7d54-472e-93c8-cfe77b35e0f5.png)

=> The URL is OK (hyperlink visible).

Thanks for help

**Comment by fanofjop** [comment]
An animated gif to explain the issue more clearly...
![tuti](https://user-images.githubusercontent.com/107914828/236911396-d808fd26-b307-4eac-b9ab-19821457ec89.gif)

**Comment by Daeraxa** [comment]
I guess this is because in one example you are effectively only copying the text and in the other you are selecting the entire html element. It would be interesting to see what the raw data stored in the clipboard actually is.

**Comment by fanofjop** [comment]
@ Daeraxa
Yes it's exactly that.
And thanks to you I made a supposition that is correct. Usually the paragraph format indications are at the end of the previous paragraph with a special mark (like in text editors). In Joplin the mark is invisible but it exists and had consequences...
If you copy the mark (with mouse), things are like you want them. If you don't copy it : the joplin editor behave strangely.
Example :
![image](https://github.com/laurent22/joplin/assets/107914828/aa7afabc-1df5-42fe-940c-e82f15a8ed88)
=> Here I only select the 2 lines from end ("?" caracter) to begining ("B" character) and then use DELETE. The result is not what it's supposed to be : the first line is emptied but not deleted.
![image](https://github.com/laurent22/joplin/assets/107914828/a82a7d51-2a5f-41aa-b8f0-5430b699b0e5)
=>Here I do nearly the same thing but I select from end ("?" caracter) to the end of previous line (just after the "o" caracter of "Hello"). Now the delete is correct !!!
So obviously I copied something that is invisible but interact with the editor.
